lib/matplotlib/hatch.py

```diff
@@ -155,6 +155,7 @@ class SmallFilledCircles(SmallCircles):
 
     def __init__(self, hatch, density):
         self.num_rows = (hatch.count('.')) * density
+        # Not super().__init__!
         Circles.__init__(self, hatch, density)
```

Comment:
I'm a bit confused why this derives from `SmallCircles`, but overrides _everything_ about it, instead of deriving from `Circles`, but I didn't want to change the inheritance tree here.

Reply:
Looking at the history, it has been like this since it was added in 2009. Maybe `SmallFilledCircles` was thought of as a specialization of `SmallCircles`.

I'd add 
~~~
# TODO: __init__ overrides all attributes set by SmallCircles.
# Should this class derive from Circles instead?
~~~

Reply:
Added.